src/python/pants/backend/python/lint/pylint/plugin_target_type.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.backend.python.target_types import COMMON_PYTHON_FIELDS, PythonSources
+from pants.engine.target import Dependencies, Target
+from pants.util.ordered_set import FrozenOrderedSet
+
+
+class PylintPluginSources(PythonSources):
+    required = True
+    expected_file_extensions = (".py",)
+
+
+class PylintPluginDependencies(Dependencies):
+    """Addresses to other targets that this plugin depends on.
+
+    These targets must either be third-party Python dependencies
+    (https://pants.readme.io/docs/python-third-party-dependencies) or be located within this
+    target's same directory or a subdirectory, due to restrictions with Pylint plugins.
```

Comment:
Based on our discussions, it doesn't sound like the second half of this is true: "or be located within this
    target's same directory or a subdirectory, due to restrictions with Pylint plugins": you mentioned that once the root is loaded, dependencies can be imported even with dots.

So I wonder whether the dependencies field actually needs to be subclassed here...?

Reply:
> you mentioned that once the root is loaded, dependencies can be imported even with dots.

Yes, so long as those dots live within a subdirectory. You cannot load any arbitrary source file.

If you're at `plugins/pylint.py`, you can import `plugins/util.py` and `plugins/subdir/util.py`. You cannot import `project/app.py`.

The sole reason we subclass Dependencies is to change the help message of `target types --details=pylint_source_plugin`.

Reply:
> If you're at plugins/pylint.py, you can import plugins/util.py and plugins/subdir/util.py. You cannot import project/app.py.

I think I still don't understand how that would be possible, because it feels like python's import handling would take over after the initial load, and it would inspect the PYTHONPATH to see what could be imported. *shrug.

Reply:
Nor do I do. I was quite perplexed when first encountering this, and remain perplexed. But because this is so niche, I time-boxed myself from spending even more time on this PR.